commands/args.go

```diff
@@ -171,7 +168,8 @@ func NewArgs(args []string) *Args {
 		globalFlags []string
 	)
 
-	slurpGlobalFlags(&args, &globalFlags, &noop)
+	slurpGlobalFlags(&args, &globalFlags)
+	noop = removeValue(&globalFlags, "--noop")
```

Comment:
Good idea. I've reversed the loop to handle cases where `--noop` might have been specified multiple times (an edge case, but we don't want the program to segfault).